jenkins/deployment/community/community-build-cortx.groovy

```diff
@@ -96,10 +98,64 @@ pipeline {
+        }
+        stage ('Setup K8s cluster on EC2') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'setting up K8s cluster on EC2', script: '''
+                pushd solutions/community-deploy/cloud/AWS
+                    AWS_IP=$(terraform show -json terraform.tfstate | jq .values.outputs.cortx_deploy_ip_addr.value 2>&1 | tee ip.txt)
+                    IP=$(cat ip.txt | tr -d '""')                
+                    ssh -i cortx.pem -o StrictHostKeyChecking=no centos@${IP} "pushd /home/centos/cortx-re/solutions/kubernetes && sudo ./cluster-setup.sh true"
+                    popd
+            '''
+            }
+        }
+        stage ('Deploy 1N cortx cluster') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'Deploying 1N cortx cluster on EC2', script: '''
+                pushd solutions/community-deploy/cloud/AWS
+                    AWS_IP=$(terraform show -json terraform.tfstate | jq .values.outputs.cortx_deploy_ip_addr.value 2>&1 | tee ip.txt)
+                    IP=$(cat ip.txt | tr -d '""')
+                    export SOLUTION_CONFIG_TYPE="automated"
+                    export CORTX_SERVER_IMAGE="cortx-rgw:2.0.0-0"
+                    export CORTX_DATA_IMAGE="cortx-data:2.0.0-0"
+                    export CORTX_CONTROL_IMAGE="cortx-control:2.0.0-0"
+                    export IMAGE_CLEANUP=${IMAGE_CLEANUP}    
+                    ssh -i cortx.pem -o StrictHostKeyChecking=no centos@${IP} 'pushd /home/centos/cortx-re/solutions/kubernetes && export SOLUTION_CONFIG_TYPE='"${SOLUTION_CONFIG_TYPE}"' && export IMAGE_CLEANUP='"${IMAGE_CLEANUP}"' && export CORTX_SERVER_IMAGE='"${CORTX_SERVER_IMAGE}"' && export CORTX_DATA_IMAGE='"${CORTX_DATA_IMAGE}"' && export CORTX_CONTROL_IMAGE='"${CORTX_CONTROL_IMAGE}"' && sudo env SOLUTION_CONFIG_TYPE=${SOLUTION_CONFIG_TYPE} env CORTX_SERVER_IMAGE=${CORTX_SERVER_IMAGE} env CORTX_CONTROL_IMAGE=${CORTX_CONTROL_IMAGE} env CORTX_DATA_IMAGE=${CORTX_DATA_IMAGE} env IMAGE_CLEANUP=${IMAGE_CLEANUP} ./cortx-deploy.sh --cortx-cluster'
+                    popd
+            '''
+            }
+        }
+        stage ('IO Sanity') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'IO Sanity on CORTX Cluster to validate bucket creation and object upload in deployed cluster', script: '''
+                pushd solutions/community-deploy/cloud/AWS
+                    AWS_IP=$(terraform show -json terraform.tfstate | jq .values.outputs.cortx_deploy_ip_addr.value 2>&1 | tee ip.txt)
+                    IP=$(cat ip.txt | tr -d '""')                
+                    ssh -i cortx.pem -o StrictHostKeyChecking=no centos@${IP} "pushd /home/centos/cortx-re/solutions/kubernetes && sudo ./cortx-deploy.sh --io-sanity"
+                    popd
+            '''
+            }
+        }
         stage ('Destroy AWS infrastructure') {
```

Comment:
Add `Destroy AWS infrastructure` in the `post` section so that AWS infra will be removed even if the job failed at any stage.

Reply:
Done and tested.

Job : https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/community_build_job2_/36/